src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -236,6 +230,36 @@ protected override string BuildChallengeUrl([NotNull] AuthenticationProperties p
         return QueryHelpers.AddQueryString(Options.AuthorizationEndpoint, parameters);
     }
 
+    private static void StandardizeRemoteAuthenticateQuery(HttpRequest request)
```

Comment:
Rather than make this a void that operates on the whole request, how about just passing in `request.Query` and then returning the new one to use so the callsite would be something like this? Would then be a bit clearer that it's mutating the reference it's being passed.

```csharp
request.Query = TryStandardizeRemoteAuthenticateQuery(request.Query);
```

Reply:
The point is this line, `request.QueryString = QueryString.Create(queryParams);`

That's how we do the trick to change `Query` and why I have to pass `Request`.

Reply:
I get that - I'm suggesting assigning it in the caller, rather than having a void and returning the new query in the new method.

```csharp
private static QueryString TryStandardizeRemoteAuthenticateQuery(QueryString queryString)
{
    if (...)
    {
        return QueryString.Create(queryParams);
    }

    return queryString;
}
```

Reply:
How about this, `private static bool TryStandardizeRemoteAuthenticateQuery(IQueryCollection query, out QueryString queryString)`?

```csharp
if (TryStandardizeRemoteAuthenticateQuery(Request.Query, out var queryString))
{
      Request.QueryString = queryString;
}
```